src/Uno.UI/DataBinding/BindingPropertyHelper.CacheKeys.cs

```diff
@@ -0,0 +1,153 @@
+﻿#nullable enable
+
+#if !NETFX_CORE
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.UI.Xaml;
+
+namespace Uno.UI.DataBinding;
+
+internal static partial class BindingPropertyHelper
+{
+
+	private class GetValueGetterCacheKey(Type type, string name, DependencyPropertyValuePrecedences? precedence, bool allowPrivateMembers)
+	{
+		private readonly int _hashCode =
+				type.GetHashCode()
+				^ name.GetHashCode()
+				^ (precedence is not null ? (int)precedence : 0)
```

Comment:
Since the enum contains the value `0`, I think using `-1` for null could be better to reduce hash collisions